Add chart challenge 2023 to vizlab homepage

hcorson-dosch-usgs opened the pull request:
Changes made:
-----------
Add new section to the 'Visualizations' portion of the site that includes a carousel of the 2023 chart challenge charts. **_EDIT:_** ~For now (for visual consistency) the carousel is placed inside an accordion. As we move away from the accordions we can change that (or we could drop now, would just need to style the 'Chart Challenge 2023 header' differently to  match the accordion header settings (minus the triangle))~ _It was bothering me to have such a short section in an accordion, so I took it out._

![image](https://github.com/DOI-USGS/vizlab-home/assets/54007288/132cffb9-7652-406c-84be-9b814722e389)

Fullscreen:
![image](https://github.com/DOI-USGS/vizlab-home/assets/54007288/6bbd20bc-9ab2-478f-bd6b-febdeb054ad8)


Changes:
* Add a pipeline to upload the chart challenge images to s3, in original format and as lower resolution webp images
* Add new component `CC23_Carousel.vue`, based off the [carousels in Snow-to-flow](https://github.com/DOI-USGS/snow-to-flow/blob/main/src/components/ImgCarousel.vue)
* Import VueImg in `main.js`, as in [Snow-to-flow](https://github.com/DOI-USGS/snow-to-flow/blob/main/src/main.js#L25-L27)
* Add the `CC23_Carousel.vue` to `PortfolioAccordions.vue` ~as an accordion item (for now)~
* Populate `CC23_Carousel.vue` using `src/assets/content/CC23.js`, with entries for each chart
* Construct caption for each chart in `CC23_Carousel.vue` using entry attributes and urls from `src/assets/content/CC23.js`
* Add lazy loading functions and functions to alter fullscreen carousel view and carousel caption [from Snow-to-flow](https://github.com/DOI-USGS/snow-to-flow/blob/main/src/views/Visualization.vue#L124-L214) to `Visualization.vue`

Note:
* It seems like v-img fullscreen view is maybe only sourcing the full-resolution image, not the webp? 

Other small edits
* Add title for every viz in `viz-list.csv` to fix the duplicate key '' warning
* Fix a typo in `About.vue`


Testing:
----------------------------
Before making this pull request, I:
- [X] Cleaned the code the way Vue likes it - run 'npm run lint --fix'        
- [ ] Made sure all tests run
- [ ] Ran WAVE plugin 508 compliance tool

I can confirm this has been checked on:
- [X] Chrome
- [ ] Safari
- [X] Edge
- [X] Firefox
- [ ] Samsung Internet
- [ ] Internet Explorer 11 (not supported, but still needs at least a working user redirect page)

cnell-usgs commented on src/assets/content/CC23.js:
```diff
@@ -0,0 +1,412 @@
+export default {
```
Long term we should use a similar approach to these charts and the other data viz on the site. This way is good, just will get confusing to have 2 separate tracking files.

cnell-usgs replied:
Maybe moving forward, using this approach so there's a json for each of the lanes on the portfolio (i.e. interactives, chart gallery)

hcorson-dosch-usgs replied:
yeah, I like that idea!

cnell-usgs commented on src/components/PortfolioAccordions.vue:
```diff
@@ -122,14 +129,15 @@
   import * as d3Base from 'd3';
```
same D3 comment

hcorson-dosch-usgs replied:
Here `d3` is used to read in the csv. We could add a workaround w/ `jQuery` or an npm module, but I think I'll leave as is for now, since we're planning to move away from the csv approach, and can drop `d3` then

cnell-usgs reviewed (commented):
This looks really good 
<img width="1544" alt="Screenshot 2023-05-31 at 7 39 42 AM" src="https://github.com/DOI-USGS/vizlab-home/assets/17803537/ecd3c705-59c6-4acd-976b-5deea4e25784">

It took a long time for one of the images to load when I clicked on it. Looking at the pipeline, it seems like we're serving up the full size images as thumbnails, is that right? Do all of the images load once any single image is clicked on?

Carousel navigation: I think some sort of side navigation is needed to show that there are more images in the carousel. You can see that by clicking the dots below, but it is pretty subtle. Something pointing to the right would help guide the user.

Placement: I think it's fine to put this at the top above the interactives, but it feels very separate from the chart gallery and as a user, I might not understand the distinction. What do you think about making this carousel a part of the chart gallery - like the chart challenge is a subset of the chart gallery? 

Contents: To simplify the chart gallery and fit better together - what about including any of the past chart challenge charts here as well?

hcorson-dosch-usgs commented on src/components/CC23_Carousel.vue:
```diff
@@ -0,0 +1,152 @@
+</template>
+
+<script>
+    import * as d3Base from 'd3';
```
Good thought - I've dropped it and used vanilla JavaScript to sort instead

hcorson-dosch-usgs commented:
> It took a long time for one of the images to load when I clicked on it. Looking at the pipeline, it seems like we're serving up the full size images as thumbnails, is that right? Do all of the images load once any single image is clicked on?

Agreed that something is up w/ how the images are loaded. I think the full-size images are being loaded as the thumbnails instead of the webp versions. I've created a new issue #37 to look into that in another PR

> Carousel navigation: I think some sort of side navigation is needed to show that there are more images in the carousel. You can see that by clicking the dots below, but it is pretty subtle. Something pointing to the right would help guide the user.

Oh good thought. I added the navigation controls:
![image](https://github.com/DOI-USGS/vizlab-home/assets/54007288/1e46c540-eb9d-436b-8aa3-42660f7a59f2)

> Placement: I think it's fine to put this at the top above the interactives, but it feels very separate from the chart gallery and as a user, I might not understand the distinction. What do you think about making this carousel a part of the chart gallery - like the chart challenge is a subset of the chart gallery?

I've moved it into the chart gallery, as a subsection:

![image](https://github.com/DOI-USGS/vizlab-home/assets/54007288/17cb2880-07a4-4605-bd41-784d9f502f91)

> Contents: To simplify the chart gallery and fit better together - what about including any of the past chart challenge charts here as well?

We could do this - I think we'd want to modify the captions to something like '2023 contribution for ....'. Only thing is, I'm pretty sure there are no charts in the chart gallery from 2022, so right now it would just be adding these 5:
![image](https://github.com/DOI-USGS/vizlab-home/assets/54007288/b2d220d6-edc8-41e5-9317-be70e3595403)

cnell-usgs commented:
This looks good to merge.

We should add the 2022 charts into the chart gallery also, but that can happen later. I think it would also be great to adapt use the carousel in some other places - like for river conditions and flow tiles and snow tiles.

hcorson-dosch-usgs commented:
> We should add the 2022 charts into the chart gallery also, but that can happen later. I think it would also be great to adapt use the carousel in some other places - like for river conditions and flow tiles and snow tiles.

Agreed! I'll add some issues now
